### ISSUE TITLE
Angular 8 HttpInterceptor giving error next.handle() is not a function

### ISSUE THREAD

**Issue by sajesh1985** [angular/angular#33914]
I am upgrading my angular 5 application to angular 8, I am creating HttpInterceptor but i am getting error ERROR next.handle is not a function (core.js).I checked HttpClientModule is loaded only once. Not sure what is the issue.

Can you please help me to fix this.

 import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Router } from '@angular/router';
 import { Observable, throwError as observableThrowError } from 'rxjs';
 import { catchError } from 'rxjs/operators';

 import { AppService } from './services/app.service';

 @Injectable(
 {
  providedIn:'root'
 })
 export class AuthHttpInterceptor implements HttpInterceptor {
 constructor(private authService: AppService, private router: Router) {}
intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
//const jwt = this.authService.getAuthToken();
const jwt = localStorage.getItem('jwt');
const authRequest = req.clone({ setHeaders: { authorization: `Bearer ${jwt}` },withCredentials: true 
});
return next.handle(req).pipe(
  catchError((err, caught) => {
    if (err.status === 401) {
      this.router.navigate(['/login'], {
        queryParams: { redirectUrl: this.router.routerState.snapshot.url }
      });
    }

    return new Observable<HttpEvent<any>>();
  })
);
 }
 }

**Comment by GustavoJobstraibizer** [comment]
Hello @sajesh1985 i've separated logic of error interceptor in other interceptor file. 


HttpInterceptor
---
```js
constructor(private authService: AppService, private router: Router) {}
       intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
        //const jwt = this.authService.getAuthToken();
        const jwt = localStorage.getItem('jwt');
        const authRequest = req.clone({ setHeaders: { authorization: `Bearer ${jwt}` },withCredentials: true
        });

        return next.handle(req);
    }
```

ErrorInterceptor
---

```js

import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { IIdentityError } from './../_interface/identity-error.interface';
import { AppService } from './services/app.service';
import { Router } from '@angular/router';

@Injectable()
export class ErrorInterceptor implements HttpInterceptor {

    constructor(private authService: AppService, private router: Router) {}

    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
        return next.handle(request).pipe(catchError(err => {
            switch (err.status) {
                case 400:
                    // do something
                case 401:
                    this.router.navigate(['/login'], {
                      queryParams: { redirectUrl: this.router.routerState.snapshot.url }
                    });
                case 404:
                    // do something
                    break;
                case 500:
                    // do something
                    break;
            }

            const error = err;
            return throwError(error);
        }));
    }
}

```
because do you can treat different status codes in this file and the logic keep separated. Try this.
Hope I helped you.

**Comment by alxhub** [comment]
Hello @sajesh1985,

Unfortunately we will not be able to make progress on this issue without a Stackblitz or other reproduction clearly demonstrating the error.

**Comment by pkozlowski-opensource** [comment]
I'm sorry but we can't reproduce the problem following the instructions you provided.

If the problem still exists in your application please [open a new issue](https://github.com/angular/angular/issues/new/choose) and follow the instructions in the issue template.
